modules/dmrpp_module/UnsupportedTypeException.h

```diff
@@ -0,0 +1,33 @@
+//
+// Created by ndp on 8/18/23.
+//
+
+
+#ifndef BES_UNSUPPORTEDTYPE_H
+#define BES_UNSUPPORTEDTYPE_H
+
+#include "config.h"
+
+#include <string>
+#include <utility>
+
+namespace dmrpp {
+
+/**
+ * @brief Used to prevent the production of dmr++ files that contain unsupported data types.
+ */
+class UnsupportedTypeException : public std::exception  {
+
+private:
+    std::string d_msg;
+    UnsupportedTypeException() = delete;
```

Comment:
Not sure what the best practice behavior for children of std::exception is, but you might want to declare/define the default versions for the copy constructor and destructor. Also, I think you should move the default ctor to the public section. I'm pretty sure leaving it private will show up as a 'code smell.'

Reply:
sonar cloud could care less

I'm trying to keep the class sup simple, an d by eliminating the default constructor we also eliminate the need for setter functions for the message.

Reply:
Added these as public:
    UnsupportedTypeException(UnsupportedTypeException &) = default;
    ~UnsupportedTypeException() = default;